scripts/nft/pullAllCompressedNfts.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function fetchCompressedNfts(walletAddress: string): Promise<any[]> {
+  const url: string | undefined = process.env.RPC_URL;
+  const body = {
+    jsonrpc: '2.0',
+    id: 'fetch-solana-assets',
+    method: 'searchAssets',
+    params: {
+      ownerAddress: walletAddress,
+      tokenType: 'all',
+      displayOptions: {
+        showNativeBalance: true,
+        showInscription: true,
+        showCollectionMetadata: true,
+      },
+    },
+  };
+
+  const maxRetries: number = 3;
+  let retryCount: number = 0;
+  let delayTime: number = 1000;
+
+  while (retryCount < maxRetries) {
+    try {
+      const response = await axios.post(url!, body);
+      if (response.data && response.data.result) {
+        const items = response.data.result.items;
+        const compressedNfts = items.filter((item: any) => item.compression?.compressed);
+
+        return compressedNfts;
```

Comment:
Have a main function that calls this function and saves the output in a file, similar to other scripts.

Reply:
Done